kusto/ingest/internal/resources/resources.go

```diff
@@ -395,3 +399,26 @@ func InitBackoff() backoff.BackOff {
 	exp.Multiplier = defaultMultiplier
 	return backoff.WithMaxRetries(exp, retryCount)
 }
+
+func FetchBlobSize(fPath string, ctx context.Context, client *http.Client) (size int64, err error) {
```

Comment:
Why is this an API of resource manager?
Also mind,
* We support semicolon credentials which AzBlob does not.
* We support MI auth which can not be run locally.
* We support storages like S3 we do not calculate size to

In all these cases, need to decide if we want to proceed to streaming or fallback to queued immediately when a blob size is not provided.

Reply:
interesting - this is again relevant to all sdks and need to be discussed -

Reply:
discussed f2f